tap_google_sheets/transform.py

```diff
@@ -0,0 +1,231 @@
+def transform_file_metadata(file_metadata):
+    # Convert to dict
+    file_metadata_tf = json.loads(json.dumps(file_metadata))
+    # Remove keys
+    if file_metadata_tf.get('lastModifyingUser'):
+        file_metadata_tf['lastModifyingUser'].pop('photoLink', None)
+        file_metadata_tf['lastModifyingUser'].pop('me', None)
+        file_metadata_tf['lastModifyingUser'].pop('permissionId', None)
+    # Add record to an array of 1
+    file_metadata_arr = []
+    file_metadata_arr.append(file_metadata_tf)
+    return file_metadata_arr
+
+# Convert Excel Date Serial Number (excel_date_sn) to datetime string
+# timezone_str: defaults to UTC (which we assume is the timezone for ALL datetimes)
+def excel_to_dttm_str(excel_date_sn, timezone_str=None):
+    if not timezone_str:
+        timezone_str = 'UTC'
+    tzn = pytz.timezone(timezone_str)
+    sec_per_day = 86400
+    excel_epoch = 25569 # 1970-01-01T00:00:00Z, Lotus Notes Serial Number for Epoch Start Date
+    epoch_sec = math.floor((excel_date_sn - excel_epoch) * sec_per_day)
+    epoch_dttm = datetime(1970, 1, 1)
+    excel_dttm = epoch_dttm + timedelta(seconds=epoch_sec)
+    utc_dttm = tzn.localize(excel_dttm).astimezone(pytz.utc)
+    utc_dttm_str = strftime(utc_dttm)
+    return utc_dttm_str
+
+# transform datetime values in the sheet
+def transform_sheet_datetime_data(value, sheet_title, col_name, col_letter, row_num, col_type):
+    if isinstance(value, (int, float)):
+        return excel_to_dttm_str(value)
+    else:
+        LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+            sheet_title, col_name, col_letter, row_num, col_type))
+        return str(value)
+
+# transform date values in the sheet
+def transform_sheet_date_data(value, sheet_title, col_name, col_letter, row_num, col_type):
+    if isinstance(value, (int, float)):
+        return excel_to_dttm_str(value)[:10]
+    else:
+        LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+            sheet_title, col_name, col_letter, row_num, col_type))
+        return str(value)
+
+# transform time values in the sheet
+def transform_sheet_time_data(value, sheet_title, col_name, col_letter, row_num, col_type):
+    if isinstance(value, (int, float)):
+        try:
+            total_secs = value * 86400 # seconds in day
+            # Create string formatted like HH:MM:SS
+            col_val = str(timedelta(seconds=total_secs))
+        except ValueError:
+            col_val = str(value)
+            LOGGER.info('WARNING: POSSIBLE DATA TYPE ERROR; SHEET: {}, COL: {}, CELL: {}{}, TYPE: {}'.format(
+                sheet_title, col_name, col_letter, row_num, col_type))
+        return col_val
+    else:
+        return str(value)
+
+# transform boolean values in the sheet
+def transform_sheet_boolean_data(value, sheet_title, col_name, col_letter, row_num, col_type, row):
```

Comment:
Where is row_num being used in this function?

Reply:
Removed row_num argument from the function.